File: src/1328-Break-a-Palindrome/1328.cpp
class Solution 
{
public:
    string breakPalindrome(string p) 
    {
        for (int i = 0; i < p.size()/2; i++)
        {
            if (p[i] != 'a') 
            {
                p[i] = 'a';
                return p;
            }
        }
        p[p.size() - 1] = 'b';
        return p.size() > 1 ? p : "";
    }
};

File: src/1328-Break-a-Palindrome/1328.py
class Solution:
    def breakPalindrome(self, p: str) -> str:
        for i in range(len(p) // 2):
            if p[i] != 'a':
                return p[:i] + 'a' + p[i + 1:]
        return p[:-1] + 'b' if p[:-1] else ''